src/Memory/ClientMemoryManager.ts

```diff
@@ -34,107 +34,109 @@ export class ReadOnlyClientMemoryManager {
-    }
-
-    /**
-     * Get value of entity as string before most recent input
-     * @param entityName Name of Entity
-     */
-    public PrevEntityValueAsString(entityName: string): (string | null) {
-        return this.prevMemories.ValueAsString(entityName)
+    protected __ToString(value: string | number | boolean | object): string {
+        if (typeof value == 'object') {
+            return JSON.stringify(value);
+        }
+        return value.toString()
     }
 
     /**
-     * Get array of MemoryValues 
-     * @param entityName Name of Entity
+     * Get current value of entity 
      */
-    public EntityValues(entityName: string): MemoryValue[] {
-        return this.curMemories.Values(entityName)
+    public Get<T>(entityName: string, converter: (memoryValues: MemoryValue[]) => T): T {
+        return converter(this.curMemories.Values(entityName))
     }
 
     /**
-     * Get array of MemoryValues before most recent input as a Prebuilt Entity
-     * @param entityName Name of Entity
+     * Get value of entity before most recent input
      */
-    public PrevEntityValues(entityName: string): MemoryValue[] {
-        return this.prevMemories.Values(entityName)
+    public GetPrevious<T>(entityName: string, converter: (memoryValues: MemoryValue[]) => T): T {
+        return converter(this.prevMemories.Values(entityName))
     }
 
     /**
-     * Get entity values a comma delminated string
-     * @param entityName Name of Entity
+     * Get information about the current running session
      */
-    public EntityValueAsList(entityName: string): string[] {
-        return this.curMemories.ValueAsList(entityName)
+    public SessionInfo(): SessionInfo {
+        return this.sessionInfo;
     }
 
-    /**
-     * Get entity values before most recent input a comma delminated string
-     * @param entityName Name of Entity
-     */
-    public PrevEntityValueAsList(entityName: string): string[] {
-        return this.prevMemories.ValueAsList(entityName)
+    public AS_VALUE(memoryValues: MemoryValue[]): MemoryValue {
+        if (memoryValues.length > 0) {
+            throw new Error(CLStrings.MEMORY_MANAGER_VALUE_LIST_EXCEPTION)
+        }
+        return memoryValues[0]
     }
 
-    /**
-     * Get entity value as a number
-     * @param entityName Name of Entity
-     */
-    public EntityValueAsNumber(entityName: string): number | null {
-        return this.curMemories.ValueAsNumber(entityName)
+    public AS_VALUE_LIST(memoryValues: MemoryValue[]): MemoryValue[] {
+        return memoryValues
     }
 
-    /**
-     * Get entity value before most recent input as a number
-     * @param entityName Name of Entity
-     */
-    public PrevValueAsNumber(entityName: string): number | null {
-        return this.prevMemories.ValueAsNumber(entityName)
+    public AS_STRING(memoryValues: MemoryValue[]): string {
+        if (memoryValues.length > 0) {
+            throw new Error(CLStrings.MEMORY_MANAGER_STRING_LIST_EXCEPTION)
```

Comment:
> throw new Error(CLStrings.MEMORY_MANAGER_STRING_LIST_EXCEPTION) [](start = 11, length = 64)

should this throw in multivalue cases? thought conversion of multiple entities to a comma separated string, uses this helper function